### DISCUSSION TITLE
After saving as a new file or moving the mind map file, the picture is lost, showing that the system cannot find the specified path

### DISCUSSION THREAD

**Discussion by David3256999** [freeplane/freeplane#1373] [Help] [answered]
When I back up the mind map file and save it as a new file or move the mind map file, the picture is lost, and it shows that the system cannot find the specified path，Please tell me the correct backup or move method Thanks a lot

**Comment by nnako** [comment]
Hi @David3256999 ,

in general, pictures and other files (like PDF, MP4, ...) are NOT stored within the mindmap itself. The mindmap itself does only contain `structured textual` content which is stored with the extension `.mm`. So, every node within a mindmap can only be a `reference` to the original picture, ... stored somewhere else.

When adding a picture to your mindmap, a `link` (reference) is generated, telling Freeplane where to look for the file when a user clicks on the respective hyperlink symbol. According to your personal choice concerning the usage of `absolute` or `relative` links within the settings, those picture references can even break, depending on what you do with your mindmap:

- when you store your mindmap into another folder, `relative` references will definitely break, while absolute references should still be usable. The latter ones do not depend on the localization of your mindmap.
- when you transfer your mindmap to another physical computer (e.g. when sending it via Email to another person), not even the (relative or absolute) reference will be broken but also the picture which is linked to does not necessarily exist in the same file system as the mindmap.

There are some solutions to packing / moving mindmaps and their content to other locations:

https://github.com/freeplane/freeplane/discussions/767

One solution is to use an external tooling which collects all relevant files, stores them in a specific folder (together with the mindmap) and then changes all (absolute and relative) references within the mindmap to match the file's new locations. You can now pack (e.g. zip) that folder and move it whereever you like. The links will work for anybody using that mindmap. When you want to create working backup, this solution should do:

https://github.com/nnako/freeplane-packer

**Reply by David3256999** [reply]
非常感谢您的回答，但我认为这并不利于如今文件云存储化（跨设备处理能力）希望能引入统一的文件管理体系，而不是松散的文件关联，再次感谢您抽出宝贵的时间回答我的问题，同时请原谅我使用汉语回复

**Comment by macmarrum** [comment, marked as answer]
@David3256999 
To backup a mind map you can use **Pack and Go-Go**. It's a Freeplane add-on which adds a menu item to create a ZIP file with the mind map, packing also its images and linked mind maps (without dependencies):
- https://docs.freeplane.org/getting-started/Add-ons_(install).html#pack-and-go-go

**Reply by David3256999** [reply]
thank you very much

**Reply by beccon4** [reply]
I have PackAndGo installed and I can use it very well. But - exchanging maps with images is still complicated. (especially when the documents travel between operating systems - windows\path  vs. linux/path or macos/path)  And all have to have an extension installed to provide a basic functionality. It would be extremely nice to integrate PackAndGo into the mainline Freeplane to seamlessly save and open zipped maps with their image dependencies.

**Comment by macmarrum** [comment]
> It would be extremely nice to integrate PackAndGo into the mainline Freeplane to seamlessly save and open zipped maps with their image dependencies.

It would. For interested people, please see also: https://github.com/freeplane/freeplane/discussions/1086#discussioncomment-6223398

When a mind map uses relative paths (within a single drive letter on Windows), you can move the mm file and its associated objects around, which makes it platform-agnostic. Such a mind map (and its external objects) could be located on a shared drive and be accessed from Window or Linux/MacOS.

Please note that since #1142, Freeplane is able to save images with relative paths.
I've submitted PR #1382, which will allow [Pack and Go-Go](https://github.com/macmarrum/freeplane-PackAndGoGo/releases) to pack such mind maps too.

**Reply by EdoFro** [reply]
@macmarrum,
I discovered that using [URI.resolve()](https://docs.oracle.com/javase/8/docs/api/java/net/URI.html#resolve-java.net.URI-) solves most problems with relative vs absolute paths vs multiple drives.

**Reply by macmarrum** [reply]
@EdoFro 
That's great. I'm happy for you. I know that the relative/absolute/cross-drive-paths problem isn't an easy one.
